src/client/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -200,33 +200,39 @@ private bool RtMatchesAccount(MsalRefreshTokenCacheItem rtItem, MsalAccountCache
             return homeAccIdMatch && clientIdMatch;
         }
 
-        private IReadOnlyList<MsalRefreshTokenCacheItem> GetAllRefreshTokensWithNoLocks(bool filterByClientId, string partitionKey = null)
+        private List<MsalRefreshTokenCacheItem> GetAllRefreshTokensWithNoLocks(bool filterByClientId, string partitionKey = null)
```

Comment:
As stated above, it might be better to remove this filter from here, as it leads to misuse.